app/site/controllers.py

```diff
@@ -31,23 +31,14 @@ def index():
       200:
         description: Succesfuly loaded home page
     """
-    try:
-        if request.method == "POST":
-            encoded_token = request.form.get('encoded_token', '')
-            if encoded_token:
-                try:
-                    payload = jwt.decode(encoded_token, app.config['API_KEY'])
-                except Exception as e:
-                    app.logger.error(e)
-                    return redirect(request.headers['Host'] + '/logout')
-                user = User().get_userinfo_by_id(payload['user'])
-                if user:
-                    return render_template("dashboard.html", user=user,
-                                           title='Dashboard'), 200
-                return redirect(request.headers['Host'] + '/logout')
-        return render_template("index.html", title='Home'), 200
-    except Exception:
-        return redirect(url_for('.logout'))
+    user, exception = get_user_from_cookie()
```

Comment:
In terms of code logic I think:

* `if exception` should come first: handle the error condition immediately
* It would read better as a chained if statement so logic was super clear

```
if exception:
  return logout
elif user:
  return dashboard
else:
  return index.html